ui/app/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -0,0 +1,158 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import deepEqual from 'fast-deep-equal'
+import { PermissionPageContainerContent } from '.'
+import { PageContainerFooter } from '../../ui/page-container'
+
+export default class PermissionPageContainer extends Component {
+
+  static propTypes = {
+    approvePermissionsRequest: PropTypes.func.isRequired,
+    rejectPermissionsRequest: PropTypes.func.isRequired,
+    selectedIdentity: PropTypes.object,
+    permissionsDescriptions: PropTypes.object.isRequired,
+    domainMetadata: PropTypes.object.isRequired,
+    request: PropTypes.object,
+    redirect: PropTypes.bool,
+    permissionRejected: PropTypes.bool,
+    requestMetadata: PropTypes.object,
+    targetDomainMetadata: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    redirect: null,
+    permissionRejected: null,
+    request: {},
+    requestMetadata: {},
+    selectedIdentity: {},
+  };
+
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  };
+
+  state = {
+    selectedPermissions: this.getRequestedMethodState(
+      this.getRequestedMethodNames(this.props)
+    ),
+    selectedAccount: this.props.selectedIdentity,
```

Comment:
Is this also stored in `state` to cache the original value? I don't follow why that would be required if so :confused:

Reply:
no, this was not needed, removed in ab0d66fa1